agent/app/agent_capability.go

```diff
@@ -73,6 +73,39 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
 		supportedVersions[version] = true
 	}
 
+	capabilities = agent.appendLoggingDriverCapabilities(capabilities)
+
+	if agent.cfg.SELinuxCapable {
+		capabilities = appendNameOnlyAttribute(capabilities, capabilityPrefix+"selinux")
+	}
+	if agent.cfg.AppArmorCapable {
+		capabilities = appendNameOnlyAttribute(capabilities, capabilityPrefix+"apparmor")
+	}
+
+	if _, ok := supportedVersions[dockerclient.Version_1_19]; ok {
+		capabilities = appendNameOnlyAttribute(capabilities, capabilityPrefix+"ecr-auth")
+		capabilities = appendNameOnlyAttribute(capabilities, attributePrefix+"execution-role-ecr-pull")
+	}
+
+	capabilities = agent.appendTaskIamRoleCapabilities(capabilities, supportedVersions)
+
+	capabilities, err := agent.appendTaskCPUMemLimitCapabilities(capabilities, supportedVersions)
```

Comment:
Can you please check if a unit test already covers this? If not, can you add one?

Reply:
https://github.com/aws/amazon-ecs-agent/blob/dev/agent/app/agent_capability_test.go#L393 seems to be covering the TaskCPUMemLimit capability setting. 

However there was a bug before this fix where AWSLogsExecutionRole override setting was dependent on TaskENIPluginVersionAttribute retrieval (https://github.com/aws/amazon-ecs-agent/blob/dev/agent/app/agent_capability.go#L144), but unit tests weren't there to catch it. Maybe I should update AWSLogsExecutionRole testcase to check for no dependency?

Reply:
Yeah, I think we should have something to prevent such regressions in the future.